src/tools/roc_recv/main.cpp

```diff
@@ -148,6 +148,17 @@ int main(int argc, char** argv) {
     core::BufferPool<audio::sample_t> sample_buffer_pool(allocator, MaxFrameSize, 1);
     packet::PacketPool packet_pool(allocator, 1);
 
+    sndio::Player player(sample_buffer_pool, allocator, args.oneshot_flag,
+                         config.channels, config.sample_rate);
+
+    if (!player.open(args.output_arg, args.type_arg)) {
+        roc_log(LogError, "can't open output file or device: %s %s", args.output_arg,
+                args.type_arg);
+        return 1;
+    }
+
+    config.sample_rate = player.get_sample_rate();
```

Comment:
I think we should check if sample_rate is zero and exit with error in this case.

Reply:
Fixed